src/AppBundle/Controller/AbstractController.php

```diff
@@ -172,4 +174,29 @@ protected function stepRedirector()
     {
         return $this->get('step_redirector');
     }
+
+    /**
+     * Get referer, only if matching an existing route
+     *
+     * @param  Request $request
+     * @param  array   $excludedRoutes
+     * @return string  referer URL, null if not existing or inside the $excludedRoutes
+     */
+    protected function getRefererUrlSafe(Request $request, array $excludedRoutes = [])
+    {
+        $refererUrlPath = str_replace('app_dev.php/', '', parse_url($request->headers->get('referer'), \PHP_URL_PATH));
+
+        try {
+            $routeParams = $this->get('router')->match($refererUrlPath);
+        } catch (ResourceNotFoundException $e) {
+            return null;
+        }
+        $routeName = $routeParams['_route'];
+        if (in_array($routeName, $excludedRoutes)) {
+            return null;
+        }
+        unset($routeParams['_route']);
```

Comment:
_route gets otherwise passed as a param, that doesn't exist in the route itself and gets rendered as a useless param &route=<routeName>